Content.Client/Atmos/UI/GasAnalyzerWindow.xaml.cs

```diff
@@ -0,0 +1,115 @@
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.Utility;
+using Content.Shared.Temperature;
+using static Content.Shared.Atmos.Components.SharedGasAnalyzerComponent;
+
+namespace Content.Client.Atmos.UI
+{
+    /// <summary>
+    /// Gas analyzer UI.
+    /// </summary>
+    [GenerateTypedNameReferences]
+    public partial class GasAnalyzerWindow : SS14Window
+    {
+        int _lastGasCount = 0;
+
+        public GasAnalyzerWindow()
+        {
+            RobustXamlLoader.Load(this);
+        }
+        public void UpdateState(GasAnalyzerBoundUserInterfaceState state)
+        {
+            GasGrid.RemoveAllChildren();
+
+            var statusMessage = new FormattedMessage();
+            StatusLabel.SetMessage(statusMessage);
+
+            if (state.Error != null)
+            {
+                GasBar.RemoveAllChildren();
+                statusMessage.PushColor(Color.Red);
+                statusMessage.AddMarkup(Loc.GetString("gas-analyzer-window-error-text", ("errorText", state.Error)));
+                statusMessage.Pop();
+                return;
+            }
+
+            statusMessage.AddMarkup(Loc.GetString("gas-analyzer-window-pressure-text", ("pressure", $"{state.Pressure:0.##}")));
```

Comment:
```cs
using Content.Shared.Localization.Units;
```

```suggestion
            statusMessage.AddMarkup(
                Loc.GetString(
                    "gas-analyzer-window-pressure-text",
                    ("pressure", Units.Pressure.Format(state.Pressure))
                )
            );
```

Reply:
Sounds good, but there seems to be a couple of small bugs in Content.Shared.Localization.Units (which doesn't seem to be used yet): it appends the localized string identifier for the unit instead of the localized string, and a couple of those identifiers are wrong for pressure units. I'll make a separate PR for this.

Reply:
Yeah, it was written in preparation for another PR, which fell through and hasn't been rewritten yet.
Nice catch on the identifier issue. Ironically, the unit system probably ought to have some unit _tests_.

Reply:
(Originally the suffix was hard-coded, but that didn't pass review, so they were moved to localizable IDs. Guess I forgot to update `Format`.)